tests/e2e/index_hints/index_hints.py

```diff
@@ -0,0 +1,196 @@
+        " * Produce {n}",
+        " * Filter",
+        " * ScanAllByLabelPropertyValue (n :Label {id2})",
+        " * Once",
+    ]
+    expected_explain_with_hint = [
+        row.replace("(n :Label {id2})", "(n :Label {id1})") for row in expected_explain_no_hint
+    ]
+
+    explain_no_hint = [
+        row["QUERY PLAN"]
+        for row in memgraph.execute_and_fetch("EXPLAIN MATCH (n:Label) WHERE n.id2 = 3 AND n.id1 = 3 RETURN n;")
+    ]
+    explain_with_hint = [
+        row["QUERY PLAN"]
+        for row in memgraph.execute_and_fetch(
+            "EXPLAIN USING INDEX :Label(id1) MATCH (n:Label) WHERE n.id2 = 3 AND n.id1 = 3 RETURN n;"
+        )
+    ]
+
+    assert explain_no_hint == expected_explain_no_hint and explain_with_hint == expected_explain_with_hint
+
+
+def test_multiple_label_property_index_hints(memgraph):
+    memgraph.execute("FOREACH (i IN range(1, 100) | CREATE (n:Label {id1: i}));")
+    memgraph.execute("FOREACH (i IN range(1, 50) | CREATE (n:Label {id2: i % 5}));")
+    memgraph.execute("CREATE INDEX ON :Label(id0);")
+    memgraph.execute("CREATE INDEX ON :Label(id1);")
+
+    expected_explain_with_hint = [
+        " * Produce {n}",
+        " * Filter",
+        " * ScanAllByLabelPropertyValue (n :Label {id1})",
+        " * Once",
+    ]
+
+    explain_with_hint = [
+        row["QUERY PLAN"]
+        for row in memgraph.execute_and_fetch(
+            "EXPLAIN USING INDEX :Label(id0), :Label(id1) MATCH (n:Label) WHERE n.id2 = 3 AND n.id1 = 3 RETURN n;"
+        )
+    ]
+
+    assert explain_with_hint == expected_explain_with_hint
+
+
+def test_multiple_applicable_label_property_index_hints(memgraph):
+    memgraph.execute("FOREACH (i IN range(1, 100) | CREATE (n:Label {id1: i}));")
+    memgraph.execute("FOREACH (i IN range(1, 50) | CREATE (n:Label {id2: i % 5}));")
+    memgraph.execute("CREATE INDEX ON :Label(id1);")
+    memgraph.execute("CREATE INDEX ON :Label(id2);")
+
+    expected_explain_with_hint = [
+        " * Produce {n}",
+        " * Filter",
+        " * ScanAllByLabelPropertyValue (n :Label {id1})",
+        " * Once",
+    ]
+
+    explain_with_hint = [
+        row["QUERY PLAN"]
+        for row in memgraph.execute_and_fetch(
+            "EXPLAIN USING INDEX :Label(id1), :Label(id2) MATCH (n:Label) WHERE n.id2 = 3 AND n.id1 = 3 RETURN n;"
+        )
+    ]
+
+    assert explain_with_hint == expected_explain_with_hint
+
+
+def test_nonexistent_label_index(memgraph):
+    memgraph.execute("FOREACH (i IN range(1, 50) | CREATE (n:Label1:Label2 {id: i}));")
+    memgraph.execute("FOREACH (i IN range(1, 10) | CREATE (n:Label2 {id: i+50}));")
+    memgraph.execute("CREATE INDEX ON :Label1;")
+
+    try:
+        result = [
+            row for row in memgraph.execute_and_fetch("EXPLAIN USING INDEX :Label2 MATCH (n:Label1:Label2) RETURN n;")
+        ]
+    except mgclient.DatabaseError as e:
+        assert str(e) == "Index for label Label2 doesn't exist"
+        return
```

Comment:
this is too aggresive. I would rather log it to the user..

Reply:
Makes sense, thanks! I’ve fixed this.